src/algebra/order/hom/ring.lean

```diff
@@ -0,0 +1,155 @@
+  has_coe_t F (α →+*o β) :=
+⟨λ f, { to_fun := f, map_one' := map_one f, map_mul' := map_mul f, map_add' := map_add f,
+  map_zero' := map_zero f, monotone' := order_hom_class.mono f }⟩
+
+namespace order_ring_hom
+variables [ordered_semiring α] [ordered_semiring β] {f g : α →+*o β}
+
+/-- Reinterpret an ordered ring homomorphism as an order homomorphism. -/
+def to_order_hom (f : α →+*o β) : α →o β := { ..f }
+
+instance : order_ring_hom_class (α →+*o β) α β :=
+{ coe := λ f, f.to_fun,
+  coe_injective' := λ f g h, by obtain ⟨⟨_, _⟩, _⟩ := f; obtain ⟨⟨_, _⟩, _⟩ := g; congr',
+  map_mul := λ f, f.map_mul',
+  map_one := λ f, f.map_one',
+  map_add := λ f, f.map_add',
+  map_zero := λ f, f.map_zero',
+  monotone := λ f, f.monotone' }
+
+/-- Helper instance for when there's too many metavariables to apply `to_fun.to_coe_fn` directly. -/
+instance : has_coe_to_fun (α →+*o β) (λ _, α → β) := ⟨λ f, f.to_fun⟩
+
+@[simp] lemma to_fun_eq_coe : f.to_fun = (f : α → β) := rfl
+
+@[ext] lemma ext (h : ∀ a, f a = g a) : f = g := fun_like.ext f g h
+
+/-- Copy of a `order_ring_hom` with a new `to_fun` equal to the old one. Useful to fix definitional
+equalities. -/
+protected def copy (f : α →+*o β) (f' : α → β) (h : f' = f) : α →+*o β :=
+{ to_fun := f',
+  .. f.to_ring_hom.copy f' $ by { ext, exact congr_fun h _ },
+  .. f.to_order_hom.copy f' $ by { ext, exact congr_fun h _ } }
+
+variable (α)
+
+/-- The identity as an ordered ring homomorphism. -/
+protected def id : α →+*o α :=
+{ monotone' := monotone_id,
+  ..ring_hom.id _ }
+
+variable {α}
+
+@[simp] lemma id_apply (x : α) : order_ring_hom.id α x = x := rfl
+
+@[simp] lemma to_ring_hom_id : (order_ring_hom.id α).to_ring_hom = ring_hom.id α := rfl
+@[simp] lemma to_order_hom_id : (order_ring_hom.id α).to_order_hom = order_hom.id := rfl
+
+instance : inhabited (α →+*o α) := ⟨order_ring_hom.id α⟩
+
+@[simp] lemma to_ring_hom_eq_coe {f : α →+*o β} : f.to_ring_hom = f := rfl
+@[simp] lemma to_order_hom_eq_coe {f : α →+*o β} : f.to_order_hom = f := rfl
+@[simp] lemma coe_ring_hom_to_fun_eq_coe_fun {f : α →+*o β} : (f : α →+* β).to_fun = f := rfl
+@[simp] lemma coe_ring_hom_coe_fun_eq_coe_fun {f : α →+*o β} : ((f : α →+* β) : α → β) = f := rfl
+@[simp] lemma coe_rel_hom_to_fun_eq_coe_fun {f : α →+*o β} :
+  (f : ((≤) : α → α → Prop) →r ((≤) : β → β → Prop)).to_fun = f := rfl
+@[simp] lemma coe_rel_hom_coe_fun_eq_coe_fun {f : α →+*o β} :
+  ((f : ((≤) : α → α → Prop) →r ((≤) : β → β → Prop)) : α → β) = f := rfl
+@[simp]
+lemma coe_mul_hom_to_fun_eq_coe_fun {f : α →+*o β} : ((f : α →+* β) : α →* β).to_fun = f := rfl
+@[simp]
+lemma coe_mul_hom_coe_fun_eq_coe_fun {f : α →+*o β} : (((f : α →+* β) : α →* β) : α → β) = f :=
+rfl
+@[simp]
+lemma coe_add_hom_to_fun_eq_coe_fun {f : α →+*o β} : ((f : α →+* β) : α →+ β).to_fun = f := rfl
+@[simp]
+lemma coe_add_hom_coe_fun_eq_coe_fun {f : α →+*o β} : (((f : α →+* β) : α →+ β) : α → β) = f :=
+rfl
```

Comment:
Are the `to_fun` versions really useful? Since all of them have proof term `rfl`, it doesn't help in term mode, and in tactic mode you can just combine the `to_fun_eq_coe` + `coe_whatever_hom_coe_fun_eq_coe_fun`.
```suggestion
@[simp] lemma to_ring_hom_eq_coe {f : α →+*o β} : f.to_ring_hom = f := rfl
@[simp] lemma to_order_hom_eq_coe {f : α →+*o β} : f.to_order_hom = f := rfl
@[simp] lemma coe_ring_hom_coe_fun_eq_coe_fun {f : α →+*o β} : ((f : α →+* β) : α → β) = f := rfl
@[simp] lemma coe_rel_hom_coe_fun_eq_coe_fun {f : α →+*o β} :
  ((f : ((≤) : α → α → Prop) →r ((≤) : β → β → Prop)) : α → β) = f := rfl
@[simp]
lemma coe_mul_hom_coe_fun_eq_coe_fun {f : α →+*o β} : (((f : α →+* β) : α →* β) : α → β) = f :=
rfl
@[simp]
lemma coe_add_hom_coe_fun_eq_coe_fun {f : α →+*o β} : (((f : α →+* β) : α →+ β) : α → β) = f :=
rfl
```